SerialPrograms/Source/NintendoSwitch/Programs/NintendoSwitch_GameEntry.cpp

```diff
@@ -136,11 +163,31 @@ void close_game_from_home(ConsoleHandle& console, JoyconContext& context){
 
                                                         // if game initially open.  |  if game initially closed
     pbf_mash_button(context, BUTTON_X, 800ms);          // - Close game.            |  - does nothing
-    pbf_move_joystick(context, 128, 255, 100ms, 10ms);  // - Does nothing.          |  - moves selector away from the closed game to avoid opening it.
-    pbf_move_joystick(context, 128, 255, 100ms, 10ms);  // - Does nothing.          |  - Press Down a second time in case we drop one.
-    pbf_mash_button(context, BUTTON_A, 400ms);          // - Confirm close game.    |  - opens an app on the home screen (e.g. Online)
-    go_home(console, context);                          // - Does nothing.          |  - goes back to home screen.
+    context.wait_for_all_requests();
+    if (is_close_game_menu_visible(console)){  // game was initially open. 
+        console.log("Detected close game menu.");
+        pbf_mash_button(context, BUTTON_A, 400ms);

```

Comment:
Is it possible to state machine this?

**close game menu:**
Press A

**main menu:**
if (close game seen) return;
if (main menu seen twice before) return;
if (main menu seen once before)
    move joysticks, press Home.
else
    mash X